assets/js/mod/locale-date.js

```diff
@@ -0,0 +1,37 @@
+define(['jquery'], function($) {
+    $('.locale-date').each(function() {
+        var date = new Date(this.innerText);
+        this.replaceWith(date.toDateString());
+    });
+
+    $('.locale-datetime').each(function() {
+        var p = this.parentElement;
+        var date = new Date(this.innerText);
+        var local_tz = Intl.DateTimeFormat().resolvedOptions().timeZone;
+        var event_tz = this.dataset['tz'];
+
+        $('<h3>In your time zone (' + local_tz + ')</h3>').prependTo(p);
+
+        this.replaceWith(date.toLocaleDateString());
+        $('<br />').appendTo(p);
+        $('<strong>Time:<strong>').appendTo(p);
+        p.append(' ');
+        p.append(date.toLocaleTimeString());
+
+        /*
+         * On-site events have a time zone. If the user time zone is different
+         * than the event one, also display the date and time of the event from
+         * the event timme zone.
+         */
+        if (event_tz && event_tz != local_tz) {
+            $('<h3>In the event time zone (' + event_tz + ')</h3>').appendTo(p);
+            $('<strong>Date:<strong> ').appendTo(p);
+            p.append(' ');
+            p.append(date.toLocaleDateString(undefined, {timeZone: event_tz}));
```

Comment:
The first parameter allows to change the language.  By not setting it, the user locale representation will be used.

While we can hardcode it to be `en-US` and match the format used in the US on the website which is written in English for consistency, I chose not to do so because "6/1/23" hardly makes sense for me and the default for my locale, "01/06/2023", is immediately readable and understood by my brain.  My guess is that it is the same for people with other time formats, so I preferred legibility over consistency.